### DISCUSSION TITLE
JS Hooks not working on M1 Mac

### DISCUSSION THREAD

**Discussion by KieranFleckney** [pocketbase/pocketbase#4834] [Q&A] [answered]
I am using the prebuilt version of the Pocketbase 0.22.10. When running on mac (for development) no JS hook works or are executed. If you edit one, you can see Pocketbase restart due to detecting the file change. However, it doesn't work.

This simple hook should print to stdout
```js
onAfterBootstrap((e) => {
    console.log('Hello World');
});
```
But it does not.

If it try this on a laptop running windows it works fine. So I am unsure if its a arm issue or a mac issue. Any help would be great.

**Comment by ganigeorgiev** [comment, marked as answer]
I don't have access to a mac to verify it but I don't think your issue is PocketBase related as the above code doesn't have anything platform specific - it just prints to the `stdout`.

Double check that your binary actually works with the `pb_hooks` directory that you are updating and that your hook file ends with `.pb.js` (_I suspect the latter_).

To explicitly specify a hook directory you can try running:

```sh
# use "./pocketbase serve --help" to list all flags
./pocketbase serve --hooksDir='/path/to/your/pb_hooks'
```

**Reply by KieranFleckney** [reply]
@ganigeorgiev ah well i had all that stuff correct apart from the file ending. I have .ps.js 😅 instead of pb.js

When i tested on my window laptop i must of typed it correctly and my brain must of kept looking past the file ending.

Thank you for your help and wasting your time 😄
